internal/common/pid_sock_status.go

```diff
@@ -24,16 +27,23 @@ func GetStatusPath() string {
 	return filepath.Join(config.GetWorkDir(), STATUS_FILENAME)
 }
 
-func WriteAndHandleStatusFile() error {
+func WriteAndHandleStatusFile() (err error) {
 	Lock(PID_SOCK_LOCK)
-	defer Unlock(PID_SOCK_LOCK)
+	defer func() {
+		err = errors.Join(err, Unlock(PID_SOCK_LOCK))
+	}()
 
 	statusPath := GetStatusPath()
 	pid := os.Getpid()
 	var statusJson StatusInfo
 
-	if !checkAndHandleStalePid() {
-		return fmt.Errorf("Failed to check and handle the pid.")
+	ok, err := checkAndHandleStalePid()
+	if !ok {
+		if err != nil {
+			return err
+		}
+
+		return fmt.Errorf("Failed to create the status file.")
 	}
```

Comment:
Handle these in two distinct conditionals.
(This is common golang style.)

First check `(err != nil)`, then check `!ok`.